src/Stateless/DotGraph/DotGraphFormatter.cs

```diff
@@ -7,78 +7,237 @@ namespace Stateless.DotGraph
+        FormatList _clusterFormat;
+        FormatList _edgeShape;
+        FormatList _exitEdges;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public DotGraphFormatter()
+        {
+            _nodeShapeState = new FormatList()
+                .Add(new Shape(ShapeType.Mrecord))
+                .Add(new Color(HtmlColor.blue));
+            _clusterFormat = new FormatList()
+                .Add(new Label("placeholder"));
+            _edgeShape = new FormatList()
+                .Add(new Label("placeholder"))
+                .Add(new Style(ShapeStyle.solid));
+            _exitEdges = new FormatList()
+                .Add(new Label("placeholder"))
+                .Add(new Style(ShapeStyle.dotted));
+        }
+
+        // TODO: Is this the way we want to create DOT graphs?  new DotGraphFormatter().ToDotGraph(sm)?
+
         /// <summary>
-        /// Produces a DOT GraphViz graph.
+        /// A string representation of the stateRepresentation machine in the DOT graph language.
         /// </summary>
-        /// <param name="stateMachineInfo">The StateMachineInfo to be mapped.</param>
-        /// <returns>DOT GraphViz text.</returns>
-        public static string Format(StateMachineInfo stateMachineInfo)
+        /// <returns>A description of all simple source states, triggers and destination states.</returns>
+        public string ToDotGraph(StateMachineInfo machineInfo)
         {
-            var bindings = stateMachineInfo.States;
+            clusterDictionary = new Dictionary<string, string>();
+            _handledStates = new List<string>();
+            _formattedNodesList = new List<string>();
+
+            FillClusterNames(machineInfo);
 
+            string dirgraphText = $"digraph {{{System.Environment.NewLine}compound=true;{System.Environment.NewLine}rankdir=\"LR\"{System.Environment.NewLine}";
+
+            // Start with the clusters
+            foreach (StateInfo stateInfo in machineInfo.States.Where(v => v.Substates.Count() > 0))
+            {
+                dirgraphText += BuildNodesRepresentation(stateInfo);
+            }
+
+            // Next process all non-cluster elements
+            foreach (var stateRep in machineInfo.States.Where(v => v.Superstate == null && v.Substates.Count() == 0))
+            {
+                dirgraphText += BuildNodesRepresentation(stateRep);
+            }
+            // now build behaviours
+            foreach (StateInfo stateInfo in machineInfo.States)
+            {
+                var behaviours = ProcessTriggerBehaviour(stateInfo).ToArray();
+                if (behaviours.Length > 0)
+                    dirgraphText += $"{System.Environment.NewLine}{string.Join(System.Environment.NewLine, behaviours)} ";
+            }
+
+            dirgraphText += $"{System.Environment.NewLine}}}";
+            return dirgraphText;
+
+        }
+
+
+        private List<string> ProcessEntries(StateInfo stateInfo)
```

Comment:
The codebase doesn't use explicit `private` or `internal` modifiers; if you are using ReSharper, configuring code style should help get rid of these. (We need to check in a DotSettings file :-))